### DISCUSSION TITLE
It is possible to disable/block desktop APP and mobile APP ? (only web access)

### DISCUSSION THREAD

**Discussion by AlexisPiST** [dani-garcia/vaultwarden#3478] [Q&A]
Hello,

Just a simple question, it is possible to disable or block others access than web UI ?
I need this functionality to implement Vaultwarden in my company.

With Desktop APP or Mobile APP users can store password indefinitely on computer, which is a big problem (content is encrypted but still accessible) 

Thanks,

**Comment by BlackDex** [comment]
Then you also need to block the CLI ;).
A very easy way would be to detect the specific headers sent by those clients. They have the the client type/version there. Not sure from the top of my head which it is.

But if someone really want's they can bypass that of course.
Also, the web-vault works the exact same way. If i just lock the web-vault, i still have all the data, and i can login while being offline, or when the server isn't reachable. So in a browser they can also store it indefinitely.

**Comment by AlexisPiST** [comment]
Ok thanks for your response !

To get around this problem, I have activated the "Vault Timeout" policy

INSERT INTO org_policies (uuid, org_uuid, atype, enabled, data) VALUES ('17a117ff-d552-45e5-9ad9-0ca2236f9cbd', '76268ce6-692c-4a14-b868-339728a34a13', '9', '1', '{"minutes":60}');

However, do you know the grammar of the action option to be applied?

For example, forcing users to log out after timeout, which might look like this:

INSERT INTO org_policies (uuid, org_uuid, atype, enabled, data) VALUES ('17a117ff-d552-45e5-9ad9-0ca2236f9cbd', '76268ce6-692c-4a14-b868-339728a34a13', '9', '1', '{"minutes":15}, "vaultTimeoutAction": "logOut"');

**Reply by BlackDex** [reply]
it should be `{"minutes":15,"action":"logOut"}` if I'm correct.
But that only works for manager roles and lower, not admin/owner if I'm correct.
And, it probably also only works for the newer clients 2023.4.x.

**Reply by AlexisPiST** [reply]
Oh! Yeah ! It's working perfectly ! (instant response 👍)
